### PR TITLE
Redo patch /profile testing; Close #98

### LINKED ISSUE #98: Add testing for new patch router /profile
Need to add some tests for this new route

### PR DESCRIPTION
**Pull request by AlexisDougherty13**
Rewrite the testing code that got deleted accidentally due to me pushing the wrong file the other day.

### CODE REVIEW

File: backend/src/router_test.go [hunk 1]
```diff
@@ -450,11 +450,23 @@ func (suite *RouterSuite) TestProfile() {
 		suite.JSONEq(`{
 			"user": `+stringify(User{Username: "Username"})+`
 		}`, w.Body.String())
+		
+		w = PATCH("/profile", gin.H{"firstname": "Elk","lastname": "Cloner",}, cookies...)
```
Comment (WillBAnders) [inline backend/src/router_test.go:481]: We should split this off into a separate top-level functions for each method (`TestGetProfile`/`TestPatchProfile`). Some of the earlier ones do this but I guess these didn't; we can make that change here.

File: backend/src/router_test.go [hunk 2]
```diff
@@ -450,11 +450,23 @@ func (suite *RouterSuite) TestProfile() {
...
+		suite.Equal(200, w.Code)
```
Comment (WillBAnders) [inline backend/src/router_test.go:482]: Mentioned in Discord but we also need to check that the user profile was updated correctly. I think just getting `/profile` is fine for that.

**Review by WillBAnders — CHANGES_REQUESTED** [review]

File: backend/src/router_test.go (hunk 2 above)
Comment (AlexisDougherty13) [reply backend/src/router_test.go:482]: Added it

File: backend/src/router_test.go (hunk 1 above)
Comment (AlexisDougherty13) [reply backend/src/router_test.go:481]: I split it up

**Pull request merged**